src/hotspot/os_cpu/linux_zero/os_linux_zero.cpp

```diff
@@ -288,16 +386,27 @@ size_t os::current_stack_size() {
 /////////////////////////////////////////////////////////////////////////////
 // helper functions for fatal error handler
 
-void os::print_context(outputStream* st, const void* context) {
-  ShouldNotCallThis();
+void os::print_context(outputStream* st, const void* ucVoid) {
+  st->print_cr("No context information.");
 }
 
-void os::print_tos_pc(outputStream *st, const void *context) {
-  ShouldNotCallThis();
+void os::print_tos_pc(outputStream *st, const void* ucVoid) {
+  const ucontext_t* uc = (const ucontext_t*)ucVoid;
+
+  address sp = (address)os::Linux::ucontext_get_sp(uc);
+  print_tos(st, sp);
+  st->cr();
+
+  // Note: it may be unsafe to inspect memory near pc. For example, pc may
+  // point to garbage if entry point in an nmethod is corrupted. Leave
+  // this at the end, and hope for the best.
+  address pc = os::Posix::ucontext_get_pc(uc);
+  print_instructions(st, pc, sizeof(char));
```

Comment:
Does print_instructions not use safefetch like os::print_hex_dump does? If yes, remove comment? If no, should it?

Reply:
It does use safe fetch after some change in the past. But all other arches also have this comment, so I prefer to be consistent with them. I think this comment is the left-over from the switch to safefetch on this path. That deserves a separate RFR, I think.

Reply:
Okay